tests/test_helpers.py

```diff
@@ -66,3 +68,59 @@ def test_local_model_non_default_name(self, tmp_path):
         # with adapter name
         result = check_if_peft_model(tmp_path / "peft-gpt2-other" / "other")
         assert result is True
+
+    def test_set_adapter_scale(self):
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+
+        # Get the PeftModel
+        model = AutoPeftModelForCausalLM.from_pretrained("ybelkada/opt-350m-lora")
+        tokenizer = AutoTokenizer.from_pretrained("facebook/opt-350m")
+
+        # Prepare inputs
+        input_prompt = "Preheat the oven to 350 degrees"
+        inputs = tokenizer(input_prompt, return_tensors="pt")
+
+        # Get logits before scaling
+        with torch.no_grad():
+            inputs = inputs.to(device)
+            model = model.to(device)
+            logits_before_scaling = model(**inputs).logits
+
+        # Store the scale of the lora layers
+        scales_before_scaling = scale_from_modules(model)
+
+        with set_adapter_scale(model=model, alpha=0.5):
+            # Check for scaling
+            scales_during_scaling = scale_from_modules(model)
+            for key in scales_before_scaling.keys():
+                assert scales_before_scaling[key] != scales_during_scaling[key]
+
+            # Generate logits during scaling
+            with torch.no_grad():
+                inputs = inputs.to(device)
+                model = model.to(device)
+                logits_during_scaling = model(**inputs).logits
+
+            assert torch.allclose(logits_before_scaling, logits_during_scaling)
```

Comment:
Shouldn't the logits be different when scaled?

Reply:
Yes you are right. With a different scaling, the logits should be different.

In this colab notebook I have tried building a GPT2 model with pretrained weights and then build the LoRA version using the `get_peft_model()` API. Upon forward passing the same input, the logits of the two models are the same.

I think I am missing something. Is this intended behaviour?

[Colab Notebook](https://colab.research.google.com/gist/ariG23498/08ecf1672347ae6d031909866ad6266a/scratchpad.ipynb)

Reply:
Resolved with [this](https://github.com/huggingface/peft/pull/1951#issuecomment-2252956884) comment.